yarn-project/sequencer-client/src/publisher/sequencer-publisher.ts

```diff
@@ -614,6 +623,79 @@ export class SequencerPublisher {
     });
   }
 
+  /**
+   * Simulates the propose tx with eth_simulateV1
+   * @param args - The propose tx args
+   * @param timestamp - The timestamp to simulate proposal at
+   * @returns The simulation result
+   */
+  private async simulateProposeTx(
+    args: readonly [
+      {
+        readonly header: `0x${string}`;
+        readonly archive: `0x${string}`;
+        readonly oracleInput: {
+          readonly feeAssetPriceModifier: 0n;
+        };
+        readonly blockHash: `0x${string}`;
+        readonly txHashes: `0x${string}`[];
+      },
+      ViemSignature[],
+      `0x${string}`,
+    ],
+    timestamp: bigint,
+  ) {
+    const rollupData = encodeFunctionData({
+      abi: RollupAbi,
+      functionName: 'propose',
+      args,
+    });
+
+    const forwarderData = encodeFunctionData({
+      abi: ForwarderAbi,
+      functionName: 'forward',
+      args: [[this.rollupContract.address], [rollupData]],
+    });
+
+    const simulationResult = await this.l1TxUtils
+      .simulate(
+        {
+          to: this.getForwarderAddress().toString(),
+          data: forwarderData,
+          gas: SequencerPublisher.PROPOSE_GAS_GUESS,
+        },
+        {
+          // @note we add 1n to the timestamp because geth implementation doesn't like simulation timestamp to be equal to the current block timestamp
+          time: timestamp + 1n,
+          // @note reth should have a 30m gas limit per block but throws errors that this tx is beyond limit so we increase here
```

Comment:
Im confused about this comment, if it is already too high im not getting why doubling it helps?

Reply:
ah i see you may not be the one who wrote this comment